client/basic-authorship/src/basic_authorship.rs

```diff
@@ -302,7 +282,30 @@ where
 
         self.transaction_pool.remove_invalid(&unqueue_invalid);
 
-        let (block, storage_changes, proof) = block_builder.build()?.into_inner();
+        let (seed,inherents) = block_builder.create_inherents(inherent_data)?;
+        for inherent in  inherents {
```

Comment:
Why moved here, after `remove_invalid` method called?

Reply:
seed is calculated from the current block extrinsic. In the original implementation inherents are put into the block first and then all the transactions. But at the time when random seed inherentis is executed it needs to know seed already thats why i needed to reorder `<inherents><extrinsics>` into `<extrinsics><inherents>`

Reply:
Got it. I assume this could be altered again after implementing the signing/verifying approach for the seed.